flink/src/main/java/org/apache/iceberg/flink/data/vectorized/VectorizedFlinkOrcReaders.java

```diff
@@ -0,0 +1,278 @@
+        }
+      };
+    }
+  }
+
+  private static class LongOrcColumnVector implements org.apache.flink.table.data.vector.LongColumnVector {
+    private long batchOffsetInFile;
+
+    LongOrcColumnVector(long batchOffsetInFile) {
+      this.batchOffsetInFile = batchOffsetInFile;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return false;
+    }
+
+    @Override
+    public long getLong(int i) {
+      return batchOffsetInFile + i;
+    }
+  }
+
+  private static class StructConverter implements Converter {
+    private final Types.StructType structType;
+    private final List<Converter> fieldConverters;
+    private final Map<Integer, ?> idToConstant;
+
+    private StructConverter(Types.StructType structType, List<Converter> fieldConverters,
+                            Map<Integer, ?> idToConstant) {
+      this.structType = structType;
+      this.fieldConverters = fieldConverters;
+      this.idToConstant = idToConstant;
+    }
+
+    @Override
+    public ColumnVector convert(org.apache.orc.storage.ql.exec.vector.ColumnVector vector, int batchSize,
+                                long batchOffsetInFile) {
+      StructColumnVector structVector = (StructColumnVector) vector;
+      List<Types.NestedField> fields = structType.fields();
+      ColumnVector[] fieldVectors = new ColumnVector[fields.size()];
+      for (int pos = 0; pos < fields.size(); pos++) {
+        Types.NestedField field = fields.get(pos);
+        if (idToConstant.containsKey(field.fieldId())) {
+          fieldVectors[pos] = getConstantVectors(field.type(), idToConstant.get(field.fieldId()));
+        } else if (field.equals(MetadataColumns.ROW_POSITION)) {
+          fieldVectors[pos] = new LongOrcColumnVector(batchOffsetInFile);
+        } else {
+          fieldVectors[pos] = fieldConverters.get(pos)
+              .convert(structVector.fields[pos], batchSize, batchOffsetInFile);
+        }
+      }
+
+      return new FlinkRowColumnVector(fieldVectors, structVector);
+    }
+
+    private ColumnVector getConstantVectors(Type type, Object constant) {
+      if (type.isPrimitiveType()) {
+        Type.TypeID typeID = type.typeId();
+        switch (typeID) {
+          case INTEGER:
+          case DATE:
+          case TIME:
+            return ConstantColumnVectors.ints(constant);
+
+          case LONG:
+            return ConstantColumnVectors.longs(constant);
+
+          case BOOLEAN:
+            return ConstantColumnVectors.booleans(constant);
+
+          case DOUBLE:
+            return ConstantColumnVectors.doubles(constant);
+
+          case FLOAT:
+            return ConstantColumnVectors.floats(constant);
+
+          case DECIMAL:
+            return ConstantColumnVectors.decimals(constant);
+
+          case TIMESTAMP:
+            return ConstantColumnVectors.timestamps(constant);
+
+          case STRING:
+            return ConstantColumnVectors.bytes(constant);
```

Comment:
In this case, we should return a `ColumnVector` with the `StringData` data type,  because we will need to convert the ORC backend `bytes` to flink's `StringData` so that apache flink could recognize it as flink's `String`.

Reply:
now the interface `org.apache.flink.table.data.vector.ColumnVector` in flink do not has a  sub 
 interface `StringDataColumnVector` to return `StringData`, so I use `BytesColumnVector` to parse `String`

Reply:
Okay,  I read the  [RowDataUtil#convertConstant](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java#L57),   the `STRING`'s data type is `StringData` and currently the `ConstantBytesColumnVector` could handle the `StringData`.  So it should be OK. I will mark this conversation as closed.